pconduit/context.go

```diff
@@ -32,3 +38,50 @@ func ConnectorTokenFromContext(ctx context.Context) string {
 	}
 	return ""
 }
+
+// -- Connector ID ----------------------------------------------------------
+
+type connectorIDCtxKey struct{}
```

Comment:
From what I see the connector ID, log level and token are given to the connector in `sdk.NewSourcePlugin`. In standalone plugins, we take it from env vars, in built-in plugins we inject it directly in the function call. So that's one part of the equation.

The other part is how the connector sends the token back to Conduit in calls to connector utilities. That happens by attaching the token to the context, that's why we have `ConnectorTokenFromContext` and `ContextWithConnectorToken`.

Given what I described above, technically there's no reason to attach the connector ID and log level to the context, at least not from the perspective of the protocol. I get that we may want to make these available to the connector through the context, I'm just saying that it's a bit strange having these methods in the protocol, since they are meant specifically for the SDK and connector side of things.

Would it make sense to move these connector ID and log-level utilities to the SDK? Same goes for `Enrich`, seems like the SDK is the only place we need it. Am I missing something?

Reply:
Maybe it's best I remove those methods altogether and add them back (to the SDK) when/if we need them?

As for `Enrich`, yup, I can move it there.

Reply:
Sidenote: I think we'll already have a need for the connector ID in the source middleware when we choose a [subject](https://github.com/ConduitIO/conduit-connector-sdk/pull/152/files#diff-17031a492e804fbc3522b4eb644b29e0900e77b18fba28e5fa02b1fc3e9cd1cbR221) (it should be unique so it doesn't clash with other connectors). We can let the user choose the subject manually, but ideally we should have a sane default IMO, and the connector ID would be a good option there.